src/components/KTabs/KTabs.vue

```diff
@@ -11,21 +11,21 @@
         class="tab-item"
         :class="{ active: activeTab === tab.hash }"
       >
-        <div
+        <component
+          :is="tabComponent(tab).tag"
           :aria-controls="hidePanels ? undefined : `panel-${tab.hash}`"
           :aria-selected="hidePanels ? undefined : (activeTab === tab.hash ? 'true' : 'false')"
           class="tab-link"
-          :class="{ 'has-panels': !hidePanels, disabled: tab.disabled }"
+          :class="{ disabled: tab.disabled }"
           role="tab"
-          :tabindex="getAnchorTabindex(tab)"
-          @click.prevent="!tab.disabled ? handleTabChange(tab.hash) : undefined"
-          @keydown.enter.prevent="!tab.disabled ? handleTabChange(tab.hash) : undefined"
-          @keydown.space.prevent="!tab.disabled ? handleTabChange(tab.hash) : undefined"
+          :tabindex="!tab.disabled ? 0 : -1"
+          v-bind="tabComponent(tab).attributes"
+          @click="!tab.disabled ? handleTabChange(tab.hash) : undefined"
         >
           <slot :name="`${getTabSlotName(tab.hash)}-anchor`">
-            <span>{{ tab.title }}</span>
```

Comment:
question: why is this span tag being removed? I'm not sure I understand changing the structure.

Reply:
I decided to remove it as it's not really necessary

Reply:
This could have downstream impact that you may not be aware of. Let's leave in place.

Reply:
I believe we should discourage consumer applications from relying on the DOM structure of our core components, as this makes any upstream changes highly susceptible to breaking. It feels rather unreasonable if we can’t even implement such internal optimizations. 🫠

Reply:
I'm more advocating for not changing the DOM structure if it isn't absolutely necessary. In this particular case, the "old" functionality (i.e. non-link tabs) doesn't require the change in this PR from `div` to `button`

Reply:
I’m just being realistic 😅 

This will be exponentially more important with the new portal UI. Customers will likely be using CSS/selectors all over the place; it’s inevitable. 

Discouraging non-critical DOM structure changes is imperative for stability. 

> I decided to remove it as it's not really necessary

TBH this is not the right pattern for an OSS component library.

Reply:
Brought the `span` back. I think my prior explanation was poorly phrased, I wanted to remove that `span` to optimize DOM structure in this component but your concern is valid.